features/step_definitions/basic_steps.rb

```diff
@@ -1,9 +1,24 @@
-When(/^I click on(?: the)? #{CAPTURE_STRING}[ ]?(link|button)?$/) do |element, type|
+When(/^I click on(?: the)?( \w*)? #{CAPTURE_STRING}[ ]?(link|button)?$/) do |ordinal, element, type|
+# use 'ordinal' when selecting among links or buttons all of which
+# have the same selector (e.g., same label)
+
+  raise 'must specify link or button to use ordinal' if ordinal and !type
+  index = 0
+  if ordinal
+    case ordinal
+    when ' first' then index = 0
+    when ' second' then index = 1
+    when ' third' then index = 2
+    else
+      raise 'do not understand ordinal value'
+    end
+  end
+
```

Comment:
In `company_steps.rb` (line 76) you use a cool trick that would simplify the code here too:

e.g. `index = ordinal ? [1, 2, 3, 4, 5].send(ordinal.lstrip) : 0`

Reply:
Thanks - that cleans up a bit of ugly code nicely.